### PR TITLE
Create REPORT_METADATA onyx key to separate report loading logic from the report data

### PR DESCRIPTION
**Pull request by ospfranco**
We are trying to optimize the ReportScreen. There are some re-renders on parent components that cause a cascade of re-renders on the screen and the list of actions. This PR takes care of certain issues:

1. Turns out the items in the list are connected to the Report object, but this object has been augmented with loading states while the actions are being fetched from the backend, which causes a lot of items to unncessarily re-render themselves. I have extracted the loading state into a separate Onyx collection, which allows to keep the logic intact but gets rid of unnecessary renders on every component that is connected to a Report.

Explanation of the graph is: The first rows are the flushing of React queue, the following bars are the rendering of the bigger components ReportScreen, ReportActionsView, etc. The last row is the rendering of individual actions in the list.

You can see the entire screen is not flushed while a lot of actions are being re-render over and over again:

CleanShot 2023-08-30 at 09 18 00

After the changes, we have regained some of the actions render cycles, but more importantly the flushing happens a bit faster (navigation mounting), so the screen actually feels smoother when navigating in/out.

CleanShot 2023-08-30 at 09 18 28

2. A shared context `ReportScreenContext` was used to manipulate certain lists, and would cause parent components to re-render causing a cascade of cyclic renders. By separating the context we limit the re-renders to the part of the tree that consume the specific contexts and win back some render cycles.

here is the react devtools trace, you can see before the change there are at least 7 renders, this is also a parent component, so it triggers a cascade of re-renders on the children. Afer the change this has gone down to 3.

CleanShot 2023-08-29 at 10 16 46

CleanShot 2023-08-29 at 10 15 44

3. The functionality of marking messages as unread was using an object to get around a `useCallback` on the renderItem function of the action list not updating properly. Also causing a full list re-render. The new implementation uses a simple integer to avoid triggering re-renders when nothing has changed.

Here is the trace before and after

CleanShot 2023-08-29 at 10 21 59
CleanShot 2023-08-29 at 10 23 08

You can see a reduction of 2 renders, but again this being a parent component it triggers a re-render of the entire list elements

4. SidebarLinks was using the withReportIDContext, but it was not necessary since it only needs the reportID and that can be passed from a parent. Passing it by prop also saves some render cycles and again this is a parent component

Before:
CleanShot 2023-08-30 at 12 31 56

After:
CleanShot 2023-08-30 at 12 32 58

5. date-fns locale was being set whenever the list of report actions was being rendered, this caused the first rendered to be delayed by at least 250 ms. I'm now eagerly trying to load it after the sidebar links have been loaded and the JS context is iddle. This results in the first report window that opens to be a lot faster.

You can see the spike created by setting this locale in the DevTools:

CleanShot 2023-08-30 at 12 49 16

### Fixed Issues

$ https://github.com/Expensify/App/issues/26087

### PR Author Checklist

- [x] I linked the correct issue in the `### Fixed Issues` section above
- [x] I wrote clear testing steps that cover the changes made in this PR
    - [x] I added steps for local testing in the `Tests` section
    - [x] I added steps for the expected offline behavior in the `Offline steps` section
    - [x] I added steps for Staging and/or Production testing in the `QA steps` section
    - [x] I added steps to cover failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
    - [x] I tested this PR with a [High Traffic account](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#high-traffic-accounts) against the staging or production API to ensure there are no regressions (e.g. long loading states that impact usability).
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I ran the tests on **all platforms** & verified they passed on:
    - [x] Android / native
    - [x] Android / Chrome
    - [x] iOS / native
    - [x] iOS / Safari
    - [x] MacOS / Chrome / Safari
    - [x] MacOS / Desktop
- [x] I verified there are no console errors (if there's a console error not related to the PR, report it or open an issue for it to be fixed)
- [x] I followed proper code patterns (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`)
    - [x] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
      - [x] If any non-english text was added/modified, I verified the translation was requested/reviewed in #expensify-open-source and it was approved by an internal Expensify engineer. Link to Slack message:
    - [x] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [x] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is approved by marketing by adding the `Waiting for Copy` label for a copy review on the original GH to get the correct copy.
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I followed the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I tested other components that can be impacted by my changes (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` are working as expected)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] I verified that if a function's arguments changed that all usages have also been updated correctly
- [x] If a new component is created I verified that:
    - [x] A similar component doesn't exist in the codebase
    - [x] All props are defined accurately and each prop has a `/** comment above it */`
    - [x] The file is named correctly
    - [x] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [x] The only data being stored in the state is data necessary for rendering and nothing else
    - [x] If we are not using the full Onyx data that we loaded, I've added the proper selector in order to ensure the component only re-renders when the data it is using changes
    - [x] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [x] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [x] All JSX used for rendering exists in the render method
    - [x] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [x] If any new file was added I verified that:
    - [x] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/StyleUtils.js) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(themeColors.componentBG)`)
- [x] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] If the PR modifies a component or page that can be accessed by a direct deeplink, I verified that the code functions as expected when the deeplink is used - from a logged in and logged out account.
- [x] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [x] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.
- [x] I have checked off every checkbox in the PR author checklist, including those that don't apply to this PR.

### Results

All in all, after all these optimizations here is the navigating in/out of the screen:

https://github.com/Expensify/App/assets/1634213/31ae6848-bd43-4552-a371-2f37c2565241

### CODE REVIEW

**Comment by mountiny** [comment]
@situchan could you run the checklist on this one please? 

@deetergp feel free to review if youa re interested cc @tgolen since this is a new Pr form this which Oscar closed https://github.com/Expensify/App/pull/25948 probbly to make it cleaner with the functional component

File: src/pages/home/report/ReportActionsList.js
Comment (mountiny) [inline src/pages/home/report/ReportActionsList.js:37]:
```suggestion
    /** The report metadata loading states */
```

File: src/pages/home/ReportScreen.js
Comment (ospfranco) [reply src/pages/home/ReportScreen.js:368]: I think nothing is shown not even the header because of onyx blocking the rendering, not because this conditional changed, but I can double check

Comment (ospfranco) [reply src/pages/home/ReportScreen.js:368]: Yep, the conditional is correct on the first render of the screen, it just seems that now the actions load so fast that the skeleton is no longer rendered. But we are still working on finding more performance improvement because the screen can be mounted even faster, then the skeleton will probably start working again

**Comment by situchan** [comment]
@mountiny we're holding this until merge freeze is over, right?

**Comment by mountiny** [comment]
> @mountiny we're holding this until merge freeze is over, right?

Yes while this would be great to have, I think it has higher risk of regression slipping through the testing so I agree

**Comment by mountiny** [comment]
Running the test build here https://github.com/Expensify/App/actions/runs/6071479549

**Comment by kbecciv** [comment]
@mountiny QA team is started the special build Regression, but experienced a crash when logged in.
Would you like us to continue with Android app Only until the issue is resolved? 

https://github.com/Expensify/App/assets/93399543/6cd0cdce-6dc2-4858-8c73-5fd6f471e876

**Comment by lanitochka17** [comment]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___
### Web crashes during login
## Action Performed:
1. Go to https://26173.pr-testing.expensify.com/
2. Log in to any account

## Expected Result:
Able to login to any account

## Actual Result:
Unable to login to any account
Web crashes during login


## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [ ] Android / native
- [x] Android / Chrome
- [ ] iOS / native
- [x] iOS / Safari
- [x] MacOS / Chrome / Safari
- [ ] MacOS / Desktop

**Version Number:** 1.3.60-1 

**Reproducible in staging?:** No

**Reproducible in production?:** No

**If this was caught during regression testing, add the test name, ID and link from TestRail:**

**Email or phone of affected tester (no customers):**

**Logs:** https://stackoverflow.com/c/expensify/questions/4856

**Notes/Photos/Videos:** Any additional supporting documentation

https://github.com/Expensify/App/assets/78819774/35b86c6d-d82d-4a0d-936e-7a1a18076a74

**Expensify/Expensify Issue URL:**

**Issue reported by:** Applause - Internal Team

**Slack conversation:**

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

**Comment by lanitochka17** [comment]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___
###  New Group - When creating a group, the application is crashed
## Action Performed:
1. Open New Expensify app
2. Log in with any login
3. Press FAB
4. Select New group
5. Add any members
6. Complete the creation of the group

## Expected Result:
Creating a new group does not cause the application to crash

## Actual Result:
When creating a group, the application is crashed


## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [x] Android / native
- [ ] Android / Chrome
- [ ] iOS / native
- [ ] iOS / Safari
- [x] Windows / Chrome 
- [ ] MacOS / Desktop

**Version Number:** 1.3.60-1 PR:26173

**Reproducible in staging?:** No

**Reproducible in production?:** No

**If this was caught during regression testing, add the test name, ID and link from TestRail:**

**Email or phone of affected tester (no customers):**

**Logs:** https://stackoverflow.com/c/expensify/questions/4856

**Notes/Photos/Videos:** Any additional supporting documentation

https://github.com/Expensify/App/assets/78819774/5c0c848a-5931-42c6-98ef-42078cf5f053

**Expensify/Expensify Issue URL:**

**Issue reported by:** Applause - Internal Team

**Slack conversation:**

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

**Comment by lanitochka17** [comment]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___
###  Task - App crashes when opening unassigned task as other group member
## Action Performed:
1. Launch New Expensify app
2. User A creates a new group
3. User A creates an unassigned task in group chat with User B and User C
4. Log in as User B
5. User B opens the task in Step 3

## Expected Result:
App does not crash

## Actual Result:
App crashes


## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [x] Android / native
- [ ] Android / Chrome
- [ ] iOS / native
- [ ] iOS / Safari
- [ ] Windows / Chrome 
- [ ] MacOS / Desktop

**Version Number:** 1.3.60-1 PR:26173

**Reproducible in staging?:** No

**Reproducible in production?:** No

**If this was caught during regression testing, add the test name, ID and link from TestRail:**

**Email or phone of affected tester (no customers):**

**Logs:** https://stackoverflow.com/c/expensify/questions/4856

**Notes/Photos/Videos:** Any additional supporting documentation

https://github.com/Expensify/App/assets/78819774/1bb69a40-e2e0-4df1-b111-ee2dd6926707

**Expensify/Expensify Issue URL:**

**Issue reported by:** Applause - Internal Team

**Slack conversation:**

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

**Comment by mountiny** [comment]
@ospfranco I think this is one of the reasons behind the crashes
image

**Comment by lanitochka17** [comment]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___
###  Reply in thread unresponsive or crashes the app on threads
## Action Performed:
1. Go to https://26173.pr-testing.expensify.com/
and login
2. Open any conversation
3. Hover over a message and click on Reply on Thread
4. Repeat on a message already on a thread
5. Right click on any message click reply on thread

## Expected Result:
A new thread is started on all flows

## Actual Result:
The thread is not opened when using the hover over icon and the app/webpage crashes when using the option from right click


## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [x] Android / native
- [ ] Android / Chrome
- [ ] iOS / native
- [ ] iOS / Safari
- [x] Windows / Chrome 
- [ ] MacOS / Desktop

**Version Number:** 1.3.60-1 PR:26173

**Reproducible in staging?:** No

**Reproducible in production?:** No

**If this was caught during regression testing, add the test name, ID and link from TestRail:**

**Email or phone of affected tester (no customers):**

**Logs:** https://stackoverflow.com/c/expensify/questions/4856

**Notes/Photos/Videos:** Any additional supporting documentation

https://github.com/Expensify/App/assets/78819774/6d480bc5-bf27-4bda-9261-ce7ae56aeb5b

https://github.com/Expensify/App/assets/78819774/47ba9a41-3690-49f5-9372-4b900171e763

**Expensify/Expensify Issue URL:**

**Issue reported by:** Applause - Internal Team

**Slack conversation:**

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

**Comment by lanitochka17** [comment]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___
###   Request money modal closes when the currency is changed
## Action Performed:
1. Go to https://26173.pr-testing.expensify.com/
2. Go to + > Request money
3. Click on the currency
4. Select another currency

## Expected Result:
The request money modal does not close

## Actual Result:
The request money modal closes


## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [x] Android / native
- [ ] Android / Chrome
- [ ] iOS / native
- [ ] iOS / Safari
- [x] Windows / Chrome 
- [ ] MacOS / Desktop

**Version Number:** 1.3.60-1 PR:26173

**Reproducible in staging?:** No

**Reproducible in production?:** No

**If this was caught during regression testing, add the test name, ID and link from TestRail:**

**Email or phone of affected tester (no customers):**

**Logs:** https://stackoverflow.com/c/expensify/questions/4856

**Notes/Photos/Videos:** Any additional supporting documentation

https://github.com/Expensify/App/assets/78819774/5d6b2b55-b7ba-484d-bd24-bc5e6015986d

**Expensify/Expensify Issue URL:**

**Issue reported by:** Applause - Internal Team

**Slack conversation:**

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

**Comment by lanitochka17** [comment]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___
###  Request Money - Route data is saved when switch to Manual tab
## Action Performed:
1. Open https://26173.pr-testing.expensify.com/
2. Click on FAB button
3. Select Request Money
4. Click on Distance tab
5. Enter Start & Finish point addresses
6. Navigate to Manual tab
7. Navigate back to Distance tab

## Expected Result:
When User add routs and than switch to Manual tab - data from Distance tab should be reset

## Actual Result:
Route data is saved when switch to Manual tab


## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [ ] Android / native
- [ ] Android / Chrome
- [ ] iOS / native
- [ ] iOS / Safari
- [x] Windows / Chrome 
- [ ] MacOS / Desktop

**Version Number:** 1.3.60-1 PR:26173

**Reproducible in staging?:** No

**Reproducible in production?:** No

**If this was caught during regression testing, add the test name, ID and link from TestRail:**

**Email or phone of affected tester (no customers):**

**Logs:** https://stackoverflow.com/c/expensify/questions/4856

**Notes/Photos/Videos:** Any additional supporting documentation

https://github.com/Expensify/App/assets/78819774/e1c9b9a4-e15a-4de2-8bca-feda6a075e06

**Expensify/Expensify Issue URL:**

**Issue reported by:** Applause - Internal Team

**Slack conversation:**

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

**Comment by lanitochka17** [comment]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___
###  Request Money - Missing Map & Next button in Distance modal
## Action Performed:
1. Navigate to https://26173.pr-testing.expensify.com/
2. Click on a FAB button
3. Select Request Money
4. Navigate to Distance tab

## Expected Result:
Distance modal should contain Start & Finish points, map & Next button

## Actual Result:
Missing Map & Next button are missing in Distance modal


## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [ ] Android / native
- [x] Android / Chrome
- [ ] iOS / native
- [ ] iOS / Safari
- [x] Windows / Chrome 
- [ ] MacOS / Desktop

**Version Number:** 1.3.60-1 PR:26173

**Reproducible in staging?:** No

**Reproducible in production?:** No

**If this was caught during regression testing, add the test name, ID and link from TestRail:**

**Email or phone of affected tester (no customers):**

**Logs:** https://stackoverflow.com/c/expensify/questions/4856

**Notes/Photos/Videos:** Any additional supporting documentation

https://github.com/Expensify/App/assets/78819774/58a7f593-2100-4485-902e-b740707f6146

**Expensify/Expensify Issue URL:**

**Issue reported by:** Applause - Internal Team

**Slack conversation:**

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

**Comment by lanitochka17** [comment]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___
###  Report - No loading animation is displayed when loading a conversation
## Action Performed:
1. Open New Expensify app
2. Log in with a HIGH traffic account.
3. Disable the internet connection in the device
4. Navigate to a chat that has a very long history
5. Scroll up the conversation history

## Expected Result:
When a conversation is loaded, a loading animation is displayed and then the conversation history data is loaded online and also offline

## Actual Result:
No loading animation is displayed when loading a conversation


## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [ ] Android / native
- [ ] Android / Chrome
- [ ] iOS / native
- [ ] iOS / Safari
- [x] Windows / Chrome 
- [ ] MacOS / Desktop

**Version Number:** 1.3.60-1 PR:26173

**Reproducible in staging?:** No

**Reproducible in production?:** No

**If this was caught during regression testing, add the test name, ID and link from TestRail:**

**Email or phone of affected tester (no customers):**

**Logs:** https://stackoverflow.com/c/expensify/questions/4856

**Notes/Photos/Videos:** Any additional supporting documentation

https://github.com/Expensify/App/assets/78819774/bed0e1f6-fc94-46b0-b250-390a4ca39c4f

**Expensify/Expensify Issue URL:**

**Issue reported by:** Applause - Internal Team

**Slack conversation:**

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

**Comment by mountiny** [comment]
@ospfranco would you be able to go through the reports and verify they are fixed/ non-repro?  thanks!

**Comment by mountiny** [comment]
@fedirjh hero volunteer to tackle this one today!

File: src/pages/home/sidebar/SidebarLinks.js
Comment (ospfranco) [reply src/pages/home/sidebar/SidebarLinks.js:85]: This is a performance optimization. updateLocale doesn't really have to do anything with onyx but rather with the i18n library to load the correct localization file. Before it would run the first time the ReportScreen is mounted but it adds ~200ms delay to the animation of the screen. By running it asynchronously after sidebarlinks has been mounted (and is idle) we take advantage of an idle moment in device to load the file without interrupting the user.

File: src/pages/home/sidebar/SidebarLinksData.js
Comment (ospfranco) [reply src/pages/home/sidebar/SidebarLinksData.js:95]: Another PR somehow changed this, this call to `getTopmostReportId` was `currentReportID` I believe.

CleanShot 2023-09-14 at 15 45 13

I will remove it but it seems a lot of people want to change this somehow. On the next PR Janic also suggested replacing this with a call to Navigation. TBH I don't know which one is correct since I'm not familiar with all the pieces of the app.

File: src/pages/home/sidebar/SidebarLinks.js
Comment (ospfranco) [reply src/pages/home/sidebar/SidebarLinks.js:85]: Yeah, everytime that function is called the locale is tried to be set, tbh I didn't want to refactor larger parts of the app, I was just tasked with improving the performance. The first call does load the locale, subsequent calls do nothing. I think that should be changed in a different PR, but for now, adding that call to sidebarlinks does improve the performance the first time the chat is opened.

**Comment by ospfranco** [comment]
I don't think 2. has to do with my PR. I'm not able to reproduce it, but looking at the code it looks like @hannojg touched it last time. It's trying to access a component via ref, so it could be an issue of the component not being mounted/rendered on time.

CleanShot 2023-09-14 at 17 02 24

File: src/pages/home/ReportScreen.js
Comment (ospfranco) [reply src/pages/home/ReportScreen.js:256]: It is used for another view down in the tree. I did not want to refactor that one too.

**Comment by mountiny** [comment]
The currency selector issue is probably a react navigation patch not being applied correctly

**Comment by ospfranco** [comment]
I'm very sure the last two reported bugs have nothing to do with my changes, but it's up to you @mountiny, I can take a look tomorrow if you feel like this should be fixed as part of this PR.

**Review by fedirjh — CHANGES_REQUESTED** [review]
Let's remove the redundant code and this should be ready to get merged.

**Comment by mountiny** [comment]
@ospfranco @fedirjh tahnk you very much for a quick work, lets get this ready to get merged

**Comment by ospfranco** [comment]
Redudant code removed :)

File: src/pages/home/sidebar/SidebarLinks.js
Comment (mountiny) [reply src/pages/home/sidebar/SidebarLinks.js:85]: > can we tackle this in a separate issue ?

Yes we can

**Comment by mountiny** [comment]
@fedirjh The build should be running for final tests, we are close 🤌

**Comment by ospfranco** [comment]
Merged main to resolve conflicts

**Comment by mountiny** [comment]
> It's unclear whether this bug is related to the current PR. I was unable to reproduce it locally. It seems more like an issue with the ad-hoc build.

I also reproduced this on the adhoc build but I am not sure if this is problem with the pr or with the build

> This issue is also present on the web, but it's less noticeable due to a quick transition . This behavior seems to stem from the report data being split into two collections, and any updates to each collection trigger a re-render. It's unclear whether we can completely eliminate this issue, but I'd like to hear your thoughts on possible solutions.

This is not looking that great I agree, the updates should be right after each other right, since its successData which is applied and removes the skeleton and then the report data should be shown immediately

**Comment by ospfranco** [comment]
> > It's unclear whether this bug is related to the current PR. I was unable to reproduce it locally. It seems more like an issue with the ad-hoc build.
> 
> I also reproduced this on the adhoc build but I am not sure if this is problem with the pr or with the build

I'm not able to reproduce the issue, however nothing of my changes should affect the behavior after log in

> > This issue is also present on the web, but it's less noticeable due to a quick transition . This behavior seems to stem from the report data being split into two collections, and any updates to each collection trigger a re-render. It's unclear whether we can completely eliminate this issue, but I'd like to hear your thoughts on possible solutions.
> 
> This is not looking that great I agree, the updates should be right after each other right, since its successData which is applied and removes the skeleton and then the report data should be shown immediately

This is rather a problem with the component tree being too slow to render due to all the stuff we already discussed and should be far less noticeable or even not there at all when all the performance improvements have been merged. However, you seem to be running in simulator, performance on a release build should make this less noticeable.

**Comment by ospfranco** [comment]
Let me check if there is anything I can do about it with the current changes

**Comment by ospfranco** [comment]
Added one more conditional to keep the skeleton view mounted while the actions are being layouted. @fedirjh give it a try.

**Comment by ospfranco** [comment]
Ah, yeah, then it doesn't work. The real issue is the mounting speed of the component, everything else is a workaround. If it is really a blocker, then I suggest it is better to close this PR and wait until the onyx one is merged (should be soon), and work on https://github.com/Expensify/App/pull/26772 as it will massively improve performance on the mounting of the list component.

**Comment by mountiny** [comment]
Alright, I think this would be a step back if we would merge this and on web we would get the skeleton flicker for cached pages too (now it loads instantly).

If there is no way around that and we need to wait for the initialValues improvement, then I think lets close this PR and focus on that, I can assign you to that PR as well @fedirjh and double the price to cover for this too

**Comment by ospfranco** [comment]
Alright, closing this

**Pull request closed without merging**
